aws-predictions/src/main/java/com/amplifyframework/predictions/aws/configuration/IdentifyEntitiesConfiguration.java

```diff
@@ -0,0 +1,136 @@
+        this.collectionId = collectionId;
+        this.generalEntityDetection = generalEntityDetection;
+        this.celebrityDetectionEnabled = celebrityDetectionEnabled;
+        this.networkPolicy = networkPolicy;
+    }
+
+    /**
+     * Construct an instance of {@link IdentifyEntitiesConfiguration} from
+     * plugin configuration JSON object.
+     * @param configurationJson the plugin configuration
+     * @return the configuration for entities detection
+     * @throws JSONException if identify configuration is malformed
+     */
+    @Nullable
+    public static IdentifyEntitiesConfiguration fromJson(@NonNull JSONObject configurationJson) throws JSONException {
+        if (!configurationJson.has("identifyEntities")) {
+            return null;
+        }
+
+        // Required fields
+        JSONObject identifyEntitiesJson = configurationJson.getJSONObject("identifyEntities");
+        String celebEnabledString = identifyEntitiesJson.getString("celebrityDetectionEnabled");
+        String networkPolicyString = identifyEntitiesJson.getString("defaultNetworkPolicy");
+
+        boolean celebEnabled = Boolean.parseBoolean(celebEnabledString);
+        NetworkPolicy networkPolicy = NetworkPolicy.fromKey(networkPolicyString);
+
+        // Optional fields
+        int maxEntities;
+        boolean generalEntity;
+        String collectionId;
+        String maxEntitiesString;
+        try {
+            maxEntitiesString = identifyEntitiesJson.getString("maxEntities");
+            collectionId = identifyEntitiesJson.getString("collectionId");
+            maxEntities = Integer.parseInt(maxEntitiesString);
+            generalEntity = maxEntities < 1 || maxEntities > MAX_VALID_ENTITIES;
+        } catch (JSONException | IllegalArgumentException exception) {
+            maxEntities = 0;
+            collectionId = "";
+            generalEntity = true;
+        }
+
+        return new IdentifyEntitiesConfiguration(maxEntities, collectionId,
+                generalEntity, celebEnabled, networkPolicy);
+    }
+
+    /**
+     * Gets the max number of entities to detect per image.
+     * @return the max number of detected entities
+     */
+    public int getMaxEntities() {
+        return maxEntities;
+    }
+
+    /**
+     * Gets the collection ID to match entities against.
+     * @return the collection ID to find matching entities in
+     */
+    @NonNull
+    public String getCollectionId() {
+        return collectionId;
+    }
+
+    /**
+     * Returns true if configured to detect entities. False if
+     * scoped to detect matching entities from collection.
+     * @return true if configured for general entity detection
+     */
+    public boolean isGeneralEntityDetection() {
+        return generalEntityDetection;
+    }
+
+    /**
+     * Returns true if configured to allow celebrity detection.
+     * @return true if configured to allow celebrity detection
+     */
+    public boolean celebrityDetectionEnabled() {
```

Comment:
Naming: `isCelebritryDection...()` (with `is[A-Z]` that Kotlin likes.)

Reply:
will do